test/versioned/langchain/openai/runnables-streaming.test.js

```diff
@@ -595,3 +598,169 @@ test('streaming disabled', async (t) => {
     }
   )
 })
+
+test('ai_monitoring disabled', async (t) => {
+  t.beforeEach((ctx) => beforeEach({ enabled: true, ctx }))
+  t.afterEach((ctx) => afterEach(ctx))
+
+  await t.test(
+    'should not create llm events when `ai_monitoring.enabled` is false',
+    (t, end) => {
+      const { agent, prompt, outputParser, model } = t.nr
+      agent.config.ai_monitoring.enabled = false
+
+      helper.runInTransaction(agent, async (tx) => {
+        const input = { topic: 'Streamed' }
+
+        const chain = prompt.pipe(model).pipe(outputParser)
+        const stream = await chain.stream(input)
+        let content = ''
+        for await (const chunk of stream) {
+          content += chunk
+        }
+
+        const { streamData: expectedContent } = mockResponses.get('Streamed response')
+        assert.equal(content, expectedContent)
+        const events = agent.customEventAggregator.events.toArray()
+        assert.equal(events.length, 0, 'should not create llm events when ai_monitoring is disabled')
+
+        tx.end()
+        end()
+      })
+    }
+  )
+
+  await t.test(
+    'should not create segment when `ai_monitoring.enabled` is false',
+    (t, end) => {
+      const { agent, prompt, outputParser, model } = t.nr
+      agent.config.ai_monitoring.enabled = false
+
+      helper.runInTransaction(agent, async (tx) => {
+        const input = { topic: 'Streamed' }
+
+        const chain = prompt.pipe(model).pipe(outputParser)
+        const stream = await chain.stream(input)
+        for await (const chunk of stream) {
+          consumeStreamChunk(chunk)
+        }
+
+        const segments = tx.trace.getChildren(tx.trace.root.id)
```

Comment:
I'd instead use `findSegment` to assert it is not there. `test/lib/metrics_helper.js` has this method